packages/contractkit/src/identity/odis/phone-number-identifier.test.ts

```diff
@@ -43,18 +59,74 @@ describe(getPhoneNumberIdentifier, () => {
     fetchMock.reset()
   })
 
-  it('Retrieves a pepper correctly', async () => {
-    fetchMock.mock(endpoint, {
-      success: true,
-      combinedSignature: '0Uj+qoAu7ASMVvm6hvcUGx2eO/cmNdyEgGn0mSoZH8/dujrC1++SZ1N6IP6v2I8A',
+  describe('Retrieves a pepper correctly', () => {
+    it('Using EncryptionKeySigner', async () => {
+      fetchMock.mock(endpoint, {
+        success: true,
+        combinedSignature: '0Uj+qoAu7ASMVvm6hvcUGx2eO/cmNdyEgGn0mSoZH8/dujrC1++SZ1N6IP6v2I8A',
+      })
+
+      await expect(
+        getPhoneNumberIdentifier(mockE164Number, mockAccount, authSigner, serviceContext)
+      ).resolves.toMatchObject({
+        e164Number: mockE164Number,
+        pepper: expectedPepper,
+        phoneHash: expectedPhoneHash,
+      })
     })
 
-    await expect(
-      getPhoneNumberIdentifier(mockE164Number, mockAccount, authSigner, serviceContext)
-    ).resolves.toMatchObject({
-      e164Number: mockE164Number,
-      pepper: 'nHIvMC9B4j2+H',
-      phoneHash: '0xf3ddadd1f488cdd42b9fa10354fdcae67c303ce182e71b30855733b50dce8301',
+    it('Using CustomSigner', async () => {
+      fetchMock.mock(endpoint, {
+        success: true,
+        combinedSignature: '0Uj+qoAu7ASMVvm6hvcUGx2eO/cmNdyEgGn0mSoZH8/dujrC1++SZ1N6IP6v2I8A',
+      })
+
+      const customSignerMethod = jest.fn((bodyString) => {
+        const key = ec.keyFromPrivate(hexToBuffer(rawKey))
+        return Promise.resolve(JSON.stringify(key.sign(bodyString).toDER()))
+      })
+      const customSigner: CustomSigner = {
+        authenticationMethod: AuthenticationMethod.CUSTOM_SIGNER,
+        customSigner: customSignerMethod,
+      }
+
+      await expect(
+        getPhoneNumberIdentifier(mockE164Number, mockAccount, customSigner, serviceContext)
+      ).resolves.toMatchObject({
+        e164Number: mockE164Number,
+        pepper: expectedPepper,
+        phoneHash: expectedPhoneHash,
+      })
+
+      expect(customSignerMethod.mock.calls.length).toBe(1)
+    })
+
+    it('Preblinding the phone number', async () => {
+      fetchMock.mock(endpoint, {
+        success: true,
+        combinedSignature: '0Uj+qoAu7ASMVvm6hvcUGx2eO/cmNdyEgGn0mSoZH8/dujrC1++SZ1N6IP6v2I8A',
+      })
+
+      const blsBlindingClient = new WasmBlsBlindingClient(serviceContext.odisPubKey)
+
+      const base64PhoneNumber = Buffer.from(mockE164Number).toString('base64')
+      const base64BlindedMessage = await blsBlindingClient.blindMessage(base64PhoneNumber)
+
+      const base64BlindSig = await performGetPhoneNumberIdentifier(
+        mockAccount,
+        authSigner,
+        serviceContext,
+        base64BlindedMessage
+      )
+
+      const base64UnblindedSig = await blsBlindingClient.unblindAndVerifyMessage(base64BlindSig)
+      const sigBuf = Buffer.from(base64UnblindedSig, 'base64')
+
+      const pepper = getPepperFromThresholdSignature(sigBuf)
+      const phoneHash = getPhoneHash(sha3, mockE164Number, pepper)
```

Comment:
Feels like this part gets a bit too "in the weeds" and it lacks a bit in ergonomics. I'm gonna leave the comment here but I read through the whole PR and understood the changes a bit. I would suggest a slightly different approach. Instead of extracting `performGetPhoneNumberIdentifier` from `getPhoneNumberIdentifier` maybe we should expose these functions:

```typescript
function getPhoneNumberIdentifier(...) {
  const base64BlindedMessage = getBlindedPhoneNumber(...);
  const base64BlindSig = getBlindedPhoneNumberSignature(...);
  return getPhoneNumberIdentifierFromSignature(...); 
}

function getBlindedPhoneNumber(...) {
  // blind the phone number
}

function getBlindedPhoneNumberSignature(...) {
  // construct body and query ODIS
}

function getPhoneNumberIdentifierFromSignature(...) {
  // construct pepper and identifier from signature  
}
``` 

This way I think we get a cleaner API that we can then compose to fit our needs. Basically we're splitting the function into 3 components `blind phone`->`get signature`->`construct identifier` because we want to `blind phone` in Valora, then `get signature` in Komenci and then give back the result to Valora to `construct identifier`.

While this split is a bit more granular it will keep the logic from bleeding out of the contractkit ODIS utils because in the current split it's basically just the `getBlindedPhoneNumberSignature` step that's extracted and callable by Komenci which leaves the other two to be implemented in Valora somehow and duplicate the logic that's already in `getPhoneNumberIdentifier` which might prove problematic in the future.

Reply:
Good call out. I was debating this as well and decided against it because the un/blinding logic is concise; but you're right, it's better to handle this in one place to minimize the impact of changes going fwd.